FX: ensure node stack trace survives copying

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #69368

Summary:

Before this PR, copying a node would lose the stack trace. This PR
ensures that the stack trace is preserved across copies.

This is useful because quantization passes would like to start
allowing the user to preserve stack traces, and we use the copy
behavior.

Test Plan:

```
python test/test_fx.py TestFX.test_stack_traces
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32835248](https://our.internmc.facebook.com/intern/diff/D32835248)

vkuzo committed:
FX: ensure node stack trace survives copying

Summary:

Before this PR, copying a node would lose the stack trace. This PR
ensures that the stack trace is preserved across copies.

This is useful because quantization passes would like to start
allowing the user to preserve stack traces, and we use the copy
behavior.

Test Plan:

```
python test/test_fx.py TestFX.test_stack_traces
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32835248).

jamesr66a commented on torch/fx/graph.py:
```diff
@@ -747,6 +747,7 @@ def node_copy(self, node: Node, arg_transform: Callable[[Node], 'Argument'] = la
         assert isinstance(kwargs, dict)
         result_node = self.create_node(node.op, node.target, args, kwargs, node.name, node.type)
         result_node.meta = copy.copy(node.meta)
+        result_node.stack_trace = copy.copy(node.stack_trace)
```
Can we move the stack trace into `meta` instead?

jamesr66a reviewed (approved):
Approving, but I would encourage you to consider my comment

vkuzo committed:
Update on "FX: ensure node stack trace survives copying"


Summary:

Before this PR, copying a node would lose the stack trace. This PR
ensures that the stack trace is preserved across copies.

This is useful because quantization passes would like to start
allowing the user to preserve stack traces, and we use the copy
behavior.

Test Plan:

```
python test/test_fx.py TestFX.test_stack_traces
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32835248](https://our.internmc.facebook.com/intern/diff/D32835248)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32835248).

vkuzo committed:
Update on "FX: ensure node stack trace survives copying"


Summary:

Before this PR, copying a node would lose the stack trace. This PR
ensures that the stack trace is preserved across copies.

This is useful because quantization passes would like to start
allowing the user to preserve stack traces, and we use the copy
behavior.

Test Plan:

```
python test/test_fx.py TestFX.test_stack_traces
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32835248](https://our.internmc.facebook.com/intern/diff/D32835248)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32835248).
